sphinx/domains/std.py

```diff
@@ -1106,8 +1104,27 @@ def note_labels(self, env: "BuildEnvironment", docname: str, document: nodes.doc
                       RemovedInSphinx40Warning, stacklevel=2)
 
 
+def warn_missing_reference(app: "Sphinx", domain: Domain, node: pending_xref) -> bool:
```

Comment:
Alternatively, it is also good to add a new method to the Domain like `Domain.warn_missing_reference(node)`. I don't have a special reason to use an event for such a purpose. Okay to move the method.

Reply:
I think it is not necessary > Domain.warn....